### ISSUE TITLE
PHPStorm Plugin + Laravel Sail (Docker) Error

### ISSUE THREAD

**Issue by roberttolton** [beyondcode/tinkerwell-community#400]
Hi, love the project, and have it working in the Tinkerwell App just fine.

But I wanted to try and use the PHPStorm plugin, but the Tinkerwell output just shows:

```
Could not open input file: /opt/tinkerwell/tinker_prepared.phar
```

Within PHPStorm, for the CLI Interpreter for my project, I have PHP detected correctly and the "Connect to existing container" option checked.

What else should I be doing to make this work? Do I need to manually copy the `tinker_prepared.phar` file?

**Comment by BoskoWilkinson** [comment]
I have the same issue. Can't find a way to make it work.

**Comment by mpociot** [comment]
Yeah this is currently an open issue with the PHPStorm plugin in combination with Docker. We'll look into it.

**Comment by georanma** [comment]
@mpociot the phpstorm plugin might be a nice work around for connecting to Lando.dev based docker environments. Looking forward to seeing the resolution here.

**Comment by danielme85** [comment]
Just bought tinkerwell for this specific use case 😞  Any update on this?

**Comment by KondorB** [comment]
Same issue for me, still doesn't work. Tinkerwell is great but a pain in the butt to use if you're used to a decent autocomplete.

**Comment by noplanman** [comment]
@mpociot Is there a temporary for this?

In the lines of:
> Do I need to manually copy the tinker_prepared.phar file?

**Comment by heidbyte** [comment]
Has anyone found a workaround? :(

**Comment by noplanman** [comment]
@heidbyte Unfortunately not...

@mpociot Do you have any idea for a quick fix?

Even if it's some manual thing that needs to be done with each update, until it's fixed.

**Comment by m3rkw** [comment]
I found a very hacky workaround until this gets fixed. This works for me on Linux.

Mount your host machine's `/tmp/` directory to `/opt/` in your container. Add this to your `docker-compose.yml` file under your php container:
```
volumes:
  - /tmp/:/opt/
```

This works because Tinkerwell creates tinkerwell**x** folders on your host machine in `/tmp/` but looks for them in your container in `/opt/`, as seen in the error message:
> ```
> Could not open input file: /opt/tinkerwell/tinker_prepared.phar
> ```

**Comment by noplanman** [comment]
@m3rkw I've tried your suggestion but it doesn't work for me.

It doesn't create any `tinkerwellX` folders in my local `/tmp` folder at all, so the error message is still there.

Maybe the MacOS version saves it somewhere else?
What version of Tinkerwell are you using?

**Comment by noplanman** [comment]
@mpociot Have this been looked into at all?

Using mainly docker and PhpStorm for development, this really makes Tinkerwell unusable.
I do use the standalone app, but having it "built into" the IDE would really be great.

(the downside of closed-source 😞)

**Comment by noplanman** [comment]
So, I've found a solution for MacOS, based on what @m3rkw suggested:

```yaml
volumes:
  - '${TMPDIR}:/opt'
```

**Comment by m3rkw** [comment]
@noplanman Nice finding a solution for MacOS!
Can you try using `${TMPDIR-/tmp}` and let me know if that works for you? It should return the correct temp directory on all unix-like OS's.

**Comment by noplanman** [comment]
@m3rkw, yep, that works perfectly!

```yaml
volumes:
  - '${TMPDIR:-/tmp}:/opt'
```

**Comment by m3rkw** [comment]
Sweet, thanks @noplanman. Temporary solution for Mac and Linux below.

Mount your host machine's `/tmp/` directory to `/opt/` in your container. Add this to your `docker-compose.yml` file under your php container:

```
volumes:
  - '${TMPDIR:-/tmp}:/opt'
```

**Comment by keyboardSmasher** [comment]
For the windows users... 
```
    volumes:
      - '/mnt/c/Users/yourusernamehere/AppData/Local/Temp:/opt'
``` 

It gets it working with plain PHP, but it can't find my Laravel classes. Any fix for that?

`PHP Fatal error:  Class 'App/Models/User' not found in Psy Shell code on line 3`

**Comment by robin7331** [comment]
Any chance you have some time to look into this @mpociot ? 🤞
